### PR TITLE
Added leaderboard functionality to Tabular

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*

*Description of changes:*

Added leaderboard functionality to Tabular

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: autogluon/task/tabular_prediction/predictor.py [hunk 1]
```diff
@@ -159,9 +159,29 @@ def evaluate_predictions(self, y_true, y_pred, silent=False, auxiliary_metrics=F
...
+    def leaderboard(self, dataset=None, silent=False):
+        """
+            Output summary of information about models produced during fit() as a pandas DataFrame.
+            Includes information on test and validation scores for all models, model training times and stack levels.
+
+            Parameters
+            ----------
+            dataset: (TabularDataset object, optional)
+                This Dataset must also contain the label-column with the same column-name as specified during fit().
+                If specified, then the leaderboard returned will contain an additional column 'score_test'
```
Comment (jwmueller) [inline autogluon/task/tabular_prediction/predictor.py]: need to clarify that score_test indicates the predictive performance evaluated on the provided dataset

**Comment by szha** [comment]
Job PR-137-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-137/1/index.html

File: autogluon/task/tabular_prediction/predictor.py (hunk 1 above)
Comment (Innixma) [reply autogluon/task/tabular_prediction/predictor.py]: Good point, clarified

**Comment by zhanghang1989** [comment]
@Innixma @jwmueller I changed the PR merging rule to only require 1 approval and hope to move faster. Let me know if it is okay for you guys.

**Comment by Innixma** [comment]
Seems fine for now, once we have more contributors we may need to change it back to 2.

**Pull request merged**